src/Animals/Animal.ts

```diff
@@ -1,20 +1,23 @@
 import { AnimalNames } from '../Enums/AnimalNamesEnum';
 import { AnimalRoles } from '../Enums/AnimalRolesEnum';
-import _ from 'lodash';
+import { pull } from 'lodash';
 
 type Value = 1 | 6 | 12 | 36 | 72;
 
 export abstract class Animal {
   protected name: AnimalNames;
+  protected imagePath: string;
   protected roles: AnimalRoles[];
   protected value?: Value;
 
   constructor(
     name: AnimalNames,
+    imagePath = '',
     value?: Value,
     ...roles: AnimalRoles[]
```

Comment:
consider using additional method to adding roles (perhaps they will need to be edited), so value can be last optional argument

Reply:
There already are methods to change roles. Indeed, if we resign from roles in the constuctor, value will be the last one and there will be no need to write 'undefined', but it would be 'undefined' anyway.